raster-cli/public/PbmMonoBin.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include "PbmMono.hpp"
+
+#include <fstream>
+
+template <typename UnsignedInteger>
+unsigned char find_highest_bit_position() {
+  UnsignedInteger i = 1;
+  unsigned char position = 0;
+
+  while (i != 0) {
+    i <<= 1;
+    ++position;
+  }
+
+  return position - 1;
+}
+
+class PbmMonoBin final : public PbmMono {
+public:
+  PbmMonoBin(const std::string& file_path) : PbmMono(file_path) {}
+
+  ///  @todo rename
+  std::ostream& output(std::ostream& out) const override {
+    if (!_pixels) {
+      throw std::logic_error("file not loaded");
+    }
+
+    PbmImage::output(out);
+
+    const auto& pixels = _pixels.get();
+
+    const unsigned char default_mask = 1 << find_highest_bit_position<unsigned char>();
+
+    unsigned char buffer = 0;
+    unsigned char mask = default_mask;
+
+    for (size_t row = 0; row < get_height(); ++row) {
```

Comment:
> size_t height =  _pixels.size();
> for(size_t row = 0; row < height; ++row) {
>      size_t width = _pixels[i].size();
>      for (size_t col = 0; col <width; ++col) {
>              auto element = _pixels[row][col];
>              out.write((const char*)&element, sizeof(element));
>              if(!out) { /* do something*/}
>       }
> }

But it will be nice if we can write the whole line instead of only one byte